runtime/gc_base/ObjectAccessBarrier.hpp

```diff
@@ -267,6 +269,9 @@ class MM_ObjectAccessBarrier : public MM_BaseVirtual
 	virtual bool postObjectRead(J9VMThread *vmThread, J9Object *srcObject, fj9object_t *srcAddress);
 	virtual bool postObjectRead(J9VMThread *vmThread, J9Class *srcClass, J9Object **srcAddress);
 
+	virtual void preMountContinuation(J9VMThread *vmThread, j9object_t contObject) {}
+	virtual void postUnmountContinuation(J9VMThread *vmThread, j9object_t contObject) {}
+	virtual void registerScavenger(MM_Scavenger *scavenger) {}
```

Comment:
I would not introduce subclass specific method in a base class. This method should be only in StandardAccessBarrier and not even virtual.

Then, of course, pointer to accessBarrier must be upcasted when invoking registerScavenger. 

Prior to that we can assert 2 things are valid, to make sure that upcast is reasonable (none tried to register to wrong type of accessBarrier) :
- isStandardGC() - this is exact guard that is used when we instantiate StandardAccessBarrier
- scavengerEnabled